Firestore/core/src/firebase/firestore/remote/datastore.h

```diff
@@ -29,19 +40,45 @@ namespace remote {
 
 class Datastore {
  public:
+  Datastore(util::AsyncQueue* firestore_queue,
+            const core::DatabaseInfo& database_info);
+
+  void Shutdown();
+
+  std::unique_ptr<GrpcStream> CreateGrpcStream(absl::string_view token,
```

Comment:
Android has createWatchStream and createWriteStream

Typescript has newPersistentWatchStream and newPersistentWriteStream

We should mirror this duality here.

Reply:
However, see top-level notes: this could be a part of a GrpcConnection class and keep this mostly current form.